engine/execution/ingestion/engine_test.go

```diff
@@ -391,6 +392,53 @@ func TestExecuteOneBlock(t *testing.T) {
 	})
 }
 
+func TestBlocksArentExecutedMultipleTimes(t *testing.T) {
+	runWithEngine(t, func(ctx testingContext) {
+
+		// A <- B
+		blockA := unittest.BlockHeaderFixture()
+		blockB := unittest.ExecutableBlockFixtureWithParent(nil, &blockA)
+		blockB.StartState = unittest.StateCommitmentFixture()
+
+		// blockA's start state is its parent's state commitment,
+		// and blockA's parent has been executed.
+		commits := make(map[flow.Identifier]flow.StateCommitment)
+		commits[blockB.Block.Header.ParentID] = blockB.StartState
+		wg := sync.WaitGroup{}
+		ctx.mockStateCommitsWithMap(commits)
+
+		ctx.state.On("Sealed").Return(ctx.snapshot)
+		ctx.snapshot.On("Head").Return(&blockA, nil)
+
+		// wait finishing execution until all the blocks are sent to execution
+		wgPut := sync.WaitGroup{}
+		wgPut.Add(1)
+
+		ctx.assertSuccessfulBlockComputation(commits, func(blockID flow.Identifier, commit flow.StateCommitment) {
+			wgPut.Wait()
+			wg.Done()
+		}, blockB, unittest.IdentifierFixture(), true)
+
+		times := 4
+
+		wg.Add(1) // wait for block B to be executed
+		for i := 0; i < times; i++ {
+			err := ctx.engine.handleBlock(context.Background(), blockB.Block)
+			require.NoError(t, err)
+		}
+		wgPut.Done()
+
+		unittest.AssertReturnsBefore(t, wg.Wait, 5*time.Second)
+
+		_, more := <-ctx.engine.Done() //wait for all the blocks to be processed
+		require.False(t, more)
+
+		_, ok := commits[blockB.ID()]
```

Comment:
This validates the block has been executed, but how does it check it only executed once, and never twice?

Reply:
We mock `ComputeBlock` calls once, so additional calls will trigger the test failure.